canvas/canvas_course_usage.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import time
+import pandas as pd
+from json.decoder import JSONDecodeError
+from requests_futures.sessions import FuturesSession
+from concurrent.futures import as_completed
+import json
+logger = logging.getLogger(__name__)
+
+
+class CanvasCourseUsage:
+    def __init__(self, canvas_url, canvas_token, retry_attempts, course_ids):
+        self.canvas_url = canvas_url,
+        self.canvas_token = canvas_token,
+        self.course_ids = course_ids
+        self.retry_attempts = retry_attempts
+        self.canvas_usage_courses = []
+        self.course_retry_list = []
+        self.retry_count = 0
```

Comment:
Is this `retry_count` shared by all requests? Is it ever re-set? I think with the asynchronous approach, you need a dictionary like this:

```
{
   course_id: retry_int
}
```

Reply:
I thought you were incremented this for each time something failed, but you're only incrementing if one or more requests failed. I think that should be okay, another way of doing it.